src/thread.rs

```diff
@@ -212,6 +414,11 @@ pub struct ThreadManager<'mir, 'tcx> {
     ///
     /// Note that this vector also contains terminated threads.
     threads: IndexVec<ThreadId, Thread<'mir, 'tcx>>,
+    /// Set of threads that are currently yielding.
+    yielding_thread_set: FxHashSet<ThreadId>,
+    /// The maximum number of yields making no progress required
+    /// on all threads to report a live-lock.
```

Comment:
Wait, I thought there's some per-thread thing for how many times we go around a `yield` loop before blocking... but this seems to be something else, since the comment says it is global ("all threads")? What is this about?